lib/astutils.cpp

```diff
@@ -80,14 +81,14 @@ const Token* findAstNode(const Token* ast, const std::function<bool(const Token*
     return result;
 }
 
-static void astFlattenRecursive(const Token *tok, std::vector<const Token *> *result, const char* op, nonneg int depth = 0)
+static void astFlattenRecursive(const Token *tok, std::vector<const Token *> *result, const char* op, size_t op_len, nonneg int depth = 0)
 {
     ++depth;
     if (!tok || depth >= 100)
         return;
-    if (tok->str() == op) {
-        astFlattenRecursive(tok->astOperand1(), result, op, depth);
-        astFlattenRecursive(tok->astOperand2(), result, op, depth);
+    if (tok->str().compare(0, std::string::npos, op, op_len) == 0) {
```

Comment:
I believe the old code is simpler. I do not see that the speedup is significant enough to complicate the code.

Reply:
I didn't like it either. I will create a helper for this to make the usage easier and will see how this affects a complete run - after looking at more pressing hotspot.